R/mail.R

```diff
@@ -6,110 +6,249 @@
+
+    # Write file to disk
+    if (grepl("\\.gz$", download_url)) {
+      # Download the .gz file
+      gz_file_path <- file.path(save_folder_path, paste0('kaiaulu_', year_month_clean, '.mbox.gz'))
+      httr::GET(download_url, httr::write_disk(gz_file_path, overwrite = TRUE), httr::timeout(60))
+
+      # Unzip the file
+      gz_con <- gzfile(gz_file_path, open = "rb")
+      out_con <- file(dest, open = "wb")
+      while (TRUE) {
+        bytes <- readBin(gz_con, what = raw(), n = 1024 * 1024)
+        if (length(bytes) == 0) break
+        writeBin(bytes, out_con)
+      }
+      close(gz_con)
+      close(out_con)
+
+      # Remove the gz file after unzipping
+      file.remove(gz_file_path)
+    } else {
+      # Download the .txt file directly
+      httr::GET(download_url, httr::write_disk(dest, overwrite = TRUE), httr::timeout(60))
+    }
+
+    # Add the downloaded file to the list
+    downloaded_files <- c(downloaded_files, dest)
+  }
 
+  # Return downloaded files
+  return(downloaded_files)
 }
 
 
-#' Convert pipermail archive files (.txt and .txt.gz) into an mbox format for use with \code{\link{parse_mbox}}
-#' @param filelist A vector of pipermail archive files from \code{\link{download_pipermail}}
-#' @return Returns `output`, the name of the resulting .mbox file in the current working directory
+#' Refresh mbox files downloaded via pipermail
+#' Uses the adopted file name convention by \code{\link{download_pipermail}} to identify
+#' the latest downloaded mbox. It deletes this file, then redownloads it along with all future months
+#' up to the current real-life month.
+#' If the directory is empty, then it downloads all pipermail files (as mbox files) via \code{\link{download_pipermail}}
+#' @param mailing_list The name of the mailing list being downloaded
+#' @param start_year_month The year and month of the first file to be downloaded
+#' @param save_folder_path The folder path in which all the downloaded pipermail files will be stored
+#' @return Returns `downloaded_files`, a vector of the downloaded files in the current working directory
 #' @export
-convert_pipermail_to_mbox <- function(filelist) {
+refresh_pipermail <- function(mailing_list, start_year_month, save_folder_path) {
 
-  #at to @ replace function
-  pipermail_atreplacer <- function(string) {
+  # Create directory if it does not exist
+  if (!dir.exists(save_folder_path)) {
+    dir.create(save_folder_path, recursive = TRUE)
+  }
 
-    rstring <- sub(" at ", "@", string)
+  # Check if the folder is empty
+  files_in_folder <- list.files(save_folder_path, pattern = "kaiaulu_\\d{6}\\.mbox$")
+  if (length(files_in_folder) == 0) {
+    # If empty, download from start_year_month to the current month
+    end_year_month <- format(Sys.Date(), "%Y%m")
+    cat("Folder is empty. Downloading from", start_year_month, "to", end_year_month, "\n")
+    download_pipermail(mailing_list, start_year_month, end_year_month, save_folder_path)
+    return(NULL)
```

Comment:
Could we add an else here instead of return(NULL)? would make the code flow more clear.